moveit_ros/moveit_servo/src/collision_check.cpp

```diff
@@ -100,58 +102,68 @@ void CollisionCheck::stop()
 
 void CollisionCheck::run()
 {
-  // Update to the latest current state
-  current_state_ = planning_scene_monitor_->getStateMonitor()->getCurrentState();
-  current_state_->updateCollisionBodyTransforms();
-  collision_detected_ = false;
-
-  // Do a timer-safe distance-based collision detection
-  collision_result_.clear();
-  getLockedPlanningSceneRO()->getCollisionEnv()->checkRobotCollision(collision_request_, collision_result_,
-                                                                     *current_state_);
-  scene_collision_distance_ = collision_result_.distance;
-  collision_detected_ |= collision_result_.collision;
-  collision_result_.print();
-
-  collision_result_.clear();
-  // Self-collisions and scene collisions are checked separately so different thresholds can be used
-  getLockedPlanningSceneRO()->getCollisionEnvUnpadded()->checkSelfCollision(
-      collision_request_, collision_result_, *current_state_, getLockedPlanningSceneRO()->getAllowedCollisionMatrix());
-  self_collision_distance_ = collision_result_.distance;
-  collision_detected_ |= collision_result_.collision;
-  collision_result_.print();
-
-  velocity_scale_ = 1;
-  // If we're definitely in collision, stop immediately
-  if (collision_detected_)
+  // Update the latest parameters
+  if (servo_params_.enable_parameter_update)
   {
-    velocity_scale_ = 0;
+    servo_params_ = servo_param_listener_->get_params();
```

Comment:
How long does this take? If it's more than a few milliseconds, maybe not a good idea.

Can you put a timer around it to benchmark?

Reply:
It takes between 2 and 10 microseconds on my machine. So I think we're good.